state-chain/runtime/src/lib.rs

```diff
@@ -2094,12 +2106,107 @@ impl_runtime_apis! {
+			broker_commission: BasisPoints,
+			min_output_amount: AssetAmount,
+			retry_duration: u32,
+			boost_fee: Option<BasisPoints>,
+			affiliate_fees: Option<Affiliates<AccountId>>,
+			dca_parameters: Option<DcaParameters>,
+		) -> Result<VaultSwapDetails, DispatchErrorWithMessage> {
+			// Validate params
+			if broker_commission != 0 || affiliate_fees.map_or(false, |fees| !fees.is_empty()) {
+				return Err(Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+					pallet_cf_swapping::Error::<Runtime>::VaultSwapBrokerFeesNotSupported,
+				)));
+			}
+			pallet_cf_swapping::Pallet::<Runtime>::validate_refund_params(retry_duration)
+				.map_err(Into::<DispatchErrorWithMessage>::into)?;
+			if let Some(params) = dca_parameters.as_ref() {
+				pallet_cf_swapping::Pallet::<Runtime>::validate_dca_params(params)
+					.map_err(Into::<DispatchErrorWithMessage>::into)?;
+			}
+			if ForeignChain::from(destination_asset) != destination_address.chain() {
+				return Err(Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+					pallet_cf_swapping::Error::<Runtime>::IncompatibleDestinationAddress,
+				)));
+			}
+
+			// Encode swap
+			match ForeignChain::from(source_asset) {
+				ForeignChain::Bitcoin => {
+					let params = UtxoEncodedData {
+						output_asset: destination_asset,
+						output_address: to_encoded_address(
+							destination_address,
+							Environment::network_environment,
+						),
+						parameters: SharedCfParameters {
+							retry_duration: retry_duration.try_into().map_err(|_| {
+								Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+									pallet_cf_swapping::Error::<Runtime>::SwapRequestDurationTooLong,
+								))
+							})?,
+							min_output_amount,
+							number_of_chunks: dca_parameters
+								.as_ref()
+								.map(|params| params.number_of_chunks)
+								.unwrap_or(1)
+								.try_into()
+								.map_err(|_| {
+									Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+										pallet_cf_swapping::Error::<Runtime>::InvalidDcaParameters,
+									))
+								})?,
+							chunk_interval: dca_parameters
+								.as_ref()
+								.map(|params| params.chunk_interval)
+								.unwrap_or(2)
+								.try_into()
+								.map_err(|_| {
+									Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+										pallet_cf_swapping::Error::<Runtime>::InvalidDcaParameters,
+									))
+								})?,
+							boost_fee: boost_fee.unwrap_or_default().try_into().map_err(|_| {
+								Into::<DispatchErrorWithMessage>::into(DispatchError::from(
+									pallet_cf_swapping::Error::<Runtime>::BoostFeeTooHigh,
+								))
+							})?,
+						},
+					};
+
+					// TODO: get private channel address. For now just return the btc vault address.
+					let btc_key = pallet_cf_threshold_signature::Pallet::<Runtime, BitcoinInstance>::keys(
+						pallet_cf_threshold_signature::Pallet::<Runtime, BitcoinInstance>::current_key_epoch().unwrap()).unwrap();
```

Comment:
If you import the KeyProvider trait you can use `BitcoinThresholdSigner::active_epoch_key()` for this.

Reply:
This code will get removed when i hook it up to the private channels, so i will leave it for now.